cli/aws-cron-validator.ts

```diff
@@ -0,0 +1,109 @@
+            throw new AWSCronError(
+                `Invalid combination of day-of-month '${dayOfMonth}' and day-of-week '${dayOfWeek}'. One must be a question mark (?)`
+            );
+        }
+
+        if (!new RegExp(this.minuteRegex()).test(minute)) {
+            throw new AWSCronError(`Invalid minute value '${minute}'.`);
+        }
+        if (!new RegExp(this.hourRegex()).test(hour)) {
+            throw new AWSCronError(`Invalid hour value '${hour}'.`);
+        }
+        if (!new RegExp(this.dayOfMonthRegex()).test(dayOfMonth)) {
+            throw new AWSCronError(`Invalid day-of-month value '${dayOfMonth}'.`);
+        }
+        if (!new RegExp(this.monthRegex(), 'i').test(month)) {
+            throw new AWSCronError(`Invalid month value '${month}'.`);
+        }
+        if (!new RegExp(this.dayOfWeekRegex(), 'i').test(dayOfWeek)) {
+            throw new AWSCronError(`Invalid day-of-week value '${dayOfWeek}'.`);
+        }
+        if (!new RegExp(this.yearRegex()).test(year)) {
+            throw new AWSCronError(`Invalid year value '${year}'.`);
+        }
+
+        return expression;
+    }
+
+    // Build up regex's incrementally improving readability of the above expressions
+    private static rangeRegex(values: string): string {
+        return `(${values}|(\\*\\-${values})|(${values}\\-${values})|(${values}\\-\\*))`;
+    }
+
+    private static listRangeRegex(values: string): string {
+        const range = this.rangeRegex(values);
+        return `(${range}(\\,${range})*)`;
+    }
+
+    private static slashRegex(values: string): string {
+        const range = this.rangeRegex(values);
+        return `((\\*|${range}|${values})\\/${this.numbers})`;
+    }
+
+    private static listSlashRegex(values: string): string {
+        const slash = this.slashRegex(values);
+        const slashOrRange = `(${slash}|${this.rangeRegex(values)})`;
+        return `(${slashOrRange}(\\,${slashOrRange})*)`;
+    }
+
+    private static commonRegex(values: string): string {
+        return `(${this.listRangeRegex(values)}|\\*|${this.listSlashRegex(values)})`;
+    }
+
+    private static minuteRegex(): string {
+        return `^(${this.commonRegex(this.minuteExp)})$`;
+    }
+
+    private static hourRegex(): string {
+        return `^(${this.commonRegex(this.hourExp)})$`;
+    }
+
+    private static dayOfMonthRegex(): string {
+        return `^(${this.commonRegex(this.dayOfMonthExp)}|\\?|L|LW|${this.dayOfMonthExp}W)$`;
+    }
+
+    private static monthRegex(): string {
+        return `^(${this.commonRegex(this.monthExp)})$`;
+    }
+
+    private static dayOfWeekRegex(): string {
+        const rangeList = this.listRangeRegex(this.dayOfWeekExp);
+        return `^(${rangeList}|\\*|\\?|${this.dayOfWeekExp}L|L|L-[1-7]|${this.dayOfWeekHash()})$`;
+    }
+
+    private static yearRegex(): string {
+        return `^(${this.commonRegex(this.yearExp)})$`;
+    }
+}
```

Comment:
nit: leave an emtpy line at the end of your files - that way if you add more to it you don't get a line change on the last line, even though the content of that line won't look any different (thats why the little red icon thing appears here on github)